sdks/java/testing/nexmark/src/test/java/org/apache/beam/sdk/nexmark/queries/QueryTest.java

```diff
@@ -188,12 +188,14 @@ public void query5MatchesModelStreaming() {
 
   @Test
   @Category(NeedsRunner.class)
+  @Ignore("https://jira.apache.org/jira/browse/BEAM-7072")
```

Comment:
Did it work before without `@Ignore`?

Reply:
There was a bug (which is also fixed by this PR by throwing exception), if there is a `JOIN a > b AND c < d`, BeamSQL executes it as `JOIN a = b AND c = d`. From code you can see if there is a [AND](https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java#L499), Beam code goes to extract JOIN column without checking what's [condition](https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java#L518).

And these ignored queries are using non equi-join, which BeamSQL has never supported.

Reply:
So. Join Rel implementation is a hack and needs lots of improvement.

Reply:
Whoa that was bad

Reply:
Yikes. One thought that we can do is if there is any equi-join then we can do it first, and then do the rest as a filter operation after that. We could do this in `toPTransform` or in rewrite rules. It might be hard in rewrite rules because the transformation is not an "optimization" unless we leave it at infinite cost until it is rewritten to what we can execute. I'm not sure how easy that will be to express.

Are there tests that unimplementable joins are rejected?

Reply:
I see. I will explore a general way to supports these: equi-join, AND/OR, function call/operators, Column Ref and FieldAccess. 


There are no tests to test rejections of unsupported JOIN. I will add some.